qt/python/mantidqt/mantidqt/widgets/sliceviewer/views/dataview.py

```diff
@@ -460,8 +460,10 @@ def get_axes_limits(self):
             if self.nonorthogonal_mode:
                 inv_tr = self.nonortho_transform.inv_tr
                 # viewing axis y not aligned with plot axis
-                xmin_p, ymax_p = inv_tr(xlim[0], ylim[1])
-                xmax_p, ymin_p = inv_tr(xlim[1], ylim[0])
+                # transform top right and bottom left corner
+                xmin_p, ymin_p = inv_tr(xlim[0], ylim[0])
```

Comment:
So I think this undoes a bug I fixed in #30342 0- as a result if you soom in on the data in non-orthog view it crops it. 
For example if I zoom in on H,K = (1.5,0.5) I get this
![image](https://user-images.githubusercontent.com/55979119/202418714-6e882603-8a1d-4ac0-850b-aaa727233191.png)
Whereas we know there is data with H<1 (in particular the fake peak we made!)

Reply:
The reason I changed this is that there's a similar piece of code in SliceViewerBasePresenter.set_axes_limits.
After my change the way the code transforms the axes limits in SliceViewerDataView.get_axes_limits and SliceViewerBasePresenter.set_axes_limits is the same and this helps because when you first enter non-ortho view there is a "round trip" done between these two bits of code where the axes limits and transformed into the non-ortho system and out of it again and you end up back where you started. When you then start plotting cut lines with a surrounding box they end up where you'd expect them to be whereas before they were being plotted in the wrong place.

I've just played around with the zoom feature and I see the problem. The zoom method where you draw a box on the non-ortho view seems a bit conceptually awkward. Before my code change the way the box was interpreted was by reading the coords of the top left\bottom right corners and this ensures the max axes limits are always greater than the min when you transform them out of the non-orthogonal basis. This is perhaps a bit arbitrary though. The user could have instead expected the other two corners of the rectangle to be used (which is what my code does). Seems to me the drawing a box feature here should really be drawing a parallelogram if you want the information communicated by the user specifying a zoom to be a new H range and a new K range.

If that's a bit much to tackle here then I suggest that the way to fix this would be to add some extra code to the zoom feature rather than to revert the code change to SliceViewerDataView.get_axes_limits. SliceViewerDataView.get_axes_limits is being called in lots of situations unrelated to user zooming in\out. For example rather than directly connecting the zoomPanFinished signal to on_data_limits_changed we could add some logic in the middle to work out the new data limits based on the coords of top left\bottom right corners.